libs/vscode/nx-workspace/src/lib/get-nx-workspace.ts

```diff
@@ -61,3 +77,28 @@ function getNxCloudEnv(workspacePath: string): any {
     return {};
   }
 }
+
+export async function getCacheableOperations(): Promise<string[]> {
+  const { workspace } = await getNxWorkspace();
+  const cacheableOperations = new Set<string>();
+  for (const key in workspace.projects) {
+    if (!workspace.projects.hasOwnProperty(key)) {
+      continue;
+    }
+    const targets = workspace.projects[key].targets;
+    for (const targetKey in targets) {
+      if (!targets?.hasOwnProperty(targetKey)) {
+        continue;
+      }
+      // TODO: remove type cast after update
+      const target = targets?.[targetKey] as TargetConfiguration & {
+        cache?: boolean;
+      };
+
+      if (target?.cache) {
+        cacheableOperations.add(targetKey);
+      }
```

Comment:
`tasksRunnerOptions.default.options.cacheableOperations` is still valid. See here for how we calculate the whole list: https://github.com/nrwl/nx/blob/master/packages/nx/src/tasks-runner/run-command.ts#L501

Reply:
It's still valid but doesn't necessarily include values. In a new nx 17 workspace there's no more `cacheableOperations` so I need to get that info elsewhere.

Reply:
Yeah, but console should check both places right? This won't find any values for existing workspaces or enterprise repos that we don't migrate to the slimmer config.

Reply:
hmm good point I assumed that the values are populated in the project graph but that was totally unfounded. I can adapt this.

Reply:
They won't be in the project graph for a 16.10 or lower repo which we want to still be compatible with.